tools/autograd/gen_variable_type.py

```diff
@@ -350,6 +350,14 @@
 return c10::Dispatcher::singleton().redispatch<${ret_and_arg_types}>(${profiled_dispatch_args});
 """)
 
+
+# TraceType templates
+TRACE_DISPATCH_UNBOXED = CodeTemplate("""\
+static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::${operator_name}", "${overload_name}");
+${assign_return_values}c10::Dispatcher::singleton().redispatch<${ret_and_arg_types}>(${trace_dispatch_args});
```

Comment:
Actually, I think `redispatch` may be the wrong thing to use in this case. The idea of redispatch is it lets you bypass the current dispatch key, without actually adding it to the exclusion set. But in the case of tracing, you always want to exclude tracing before continuing on (the same way variable gets excluded before you redispatch). So you really just want to do a "normal" redispatch, in the same way that VariableType wrappers did it.

Reply:
VariableType does so via `AutoNonVariableTypeMode` + calling `at::xxx()` again - for Tracing is a combination of `setTracingState(null)` (for other ops being called) and redispatch (for root op itself here).

Do you suggest using NoTracerDispatchMode here as well and change to regular `op.call` call? (note that I won't remove `setTracingState(null)` immediately in this diff because they might still be needed for factory methods.

Reply:
@ezyang I (speculatively) changed to NoTracerDispatchMode guard + call() to replace redispatch(). It's not the "same way that VariableType wrappers did it", though - because VariableType wrapper generator has logic to make direct TypeDefault::xxx() call for certain cases, which I cannot do because I should still redispatch to VariableType kernels first.